src/main/java/org/broadinstitute/consent/http/db/DarCollectionSummaryDAO.java

```diff
@@ -139,4 +139,77 @@ List<DarCollectionSummary> getDarCollectionSummariesForSO(
   )
   List<DarCollectionSummary> getDarCollectionSummariesForResearcher(
       @Bind("userId") Integer userId);
+
+
+  @RegisterBeanMapper(value = DarCollectionSummary.class)
+  @RegisterBeanMapper(value = DarCollection.class)
+  @RegisterBeanMapper(value = Vote.class, prefix = "v")
+  @RegisterBeanMapper(value = Election.class)
+  @UseRowReducer(DarCollectionSummaryReducer.class)
+  @SqlQuery
+  (
+    "SELECT c.collection_id as dar_collection_id, c.dar_code, dar.submission_date, u.display_name as researcher_name, " +
+      "i.institution_name, e.electionid, e.status, e.datasetid, e.referenceid, v.voteid as v_vote_id, dd.dataset_id as dd_datasetid, " +
+      "v.dacuserid as v_dac_user_id, v.vote as v_vote, v.electionid as v_election_id, v.createdate as v_create_date, v.updatedate as v_update_date, v.type as v_type, " +
+      "(dar.data #>> '{}')::jsonb ->> 'projectTitle' AS name " +
+    "FROM dar_collection c " +
+    "INNER JOIN users u " +
+      "ON u.user_id = c.create_user_id " +
+    "INNER JOIN institution i " +
+      "ON i.institution_id = u.institution_id " +
+    "INNER JOIN data_access_request dar " +
+      "ON dar.collection_id = c.collection_id " +
+    "LEFT JOIN ( " +
+      "SELECT election.*, MAX(election.electionid) OVER(PARTITION BY election.referenceid, election.datasetid) AS latest " +
+      "FROM election " +
+      "WHERE LOWER(election.electiontype) = 'dataaccess' AND election.datasetid IN (<datasetIds>)" +
+    ") AS e " +
+      "ON e.referenceid = dar.reference_id " +
+    "LEFT JOIN vote v " +
+      "ON e.electionid = v.electionid " +
+    "INNER JOIN dar_dataset dd " +
+      "ON dar.reference_id = dd.reference_id " +
+    "WHERE c.collection_id= :collectionId " +
+      "AND dd.dataset_id IN (<datasetIds>) " +
+      "AND (e.latest = e.electionid OR e.electionid IS NULL) " +
+      "AND (LOWER(v.type) = 'final' OR (v.dacuserid = :currentUserId OR v.voteid IS NULL)) " +
+      "AND (LOWER(data->>'status') != 'archived' OR data->>'status' IS NULL )"
+  )
+  List<DarCollectionSummary> getDarCollectionSummaryForDACByCollectionId(
+          @Bind("currentUserId") Integer currentUserId,
+          @BindList("datasetIds") List<Integer> datasetIds,
+          @Bind("collectionId") Integer collectionId);
```

Comment:
Question, why are we returning a list instead of a single summary? We know that we should have a single result from the collection id.

Reply:
Outside of this method, I don't have a preference for what we should do at the moment - it's important to look at them in isolation. When I'm looking at a method, it has to stand on its own and make sense within its own context. `getDarCollectionSummaryForDACByCollectionId` says to me, in plain English, "Find me a single summary from a collection id". If the method does something different than that, my alarm bells go off.

Reply:
Another option is to redefine the process functions so that they take only a single summary rather than a list of summaries. You would have to iterate through the summary list outside of those process functions (for the list return, role-based queries), but it'll make the functions easier to use.

Reply:
Gotcha, makes sense!